api/src/main/java/org/openmrs/module/fhir2/providers/r4/ServiceRequestFhirResourceProvider.java

```diff
@@ -121,7 +133,45 @@ public IBundleProvider searchForServiceRequests(
 			includes = null;
 		}
 		
-		return serviceRequestService.searchForServiceRequests(patientReference, code, encounterReference,
+		hasParameter(has);
+		
+		return serviceRequestService.searchForServiceRequests(patientReference, code, has, encounterReference,
 		    participantReference, occurrence, uuid, lastUpdated, includes);
 	}
+	
+	private void hasParameter(HasParam has) {
+		List<List<IQueryParameterType>> theHasParameters = Lists.newArrayList();
```

Comment:
It seems like a bad idea to implement the `hasParameter` here. `_has` is basically asking for a JOIN, so it would be better to add this at the DAO level where we can just add the join. We also have some nifty ways to handle these embedded parameters without so many nested loops.

Reply:
I also don't see any real advantage of this over `new ArrayList<>()`...